src/test/java/com/thebois/models/beings/BeingTests.java

```diff
@@ -160,4 +183,79 @@ public void equalsReturnsFalseForNotEqualBeings(final AbstractBeing first,
         assertThat(first).isNotEqualTo(second);
     }
 
+    @Test
+    public void pawnBuildsHouses() {
+        // Arrange
+
+        final IFinder mockFinder = Mockito.mock(IFinder.class);
+        final Position position = new Position();
+        final IStructure structure = new House(position);
+        when(mockFinder.findNearestStructure(any())).thenReturn(Optional.of(structure));
+        final Pawn pawn = new Pawn(position, position, Mockito.mock(Random.class), mockFinder);
+
+        // Act
+        final float initialBuildStatus = structure.builtStatus();
+        pawn.update();
+        final float finalBuildStatus = structure.builtStatus();
```

Comment:
Here you're testing more than just `Pawn`. To isolate one from the other you should mock `IStructure` instead of use a real one, and then assert that the `Pawn` calls the `deliverItem` method of the mocked structure:

```java
// Arrange
[...]
IStructure mockStructure = Mockito.mock(IStructure.class);

// Act
pawn.update();

// Assert
verify(mockStructure, times(1)).deliverItem(any());
```